thefuck/utils.py

```diff
@@ -196,6 +196,37 @@ def _for_app(fn, command):
     return decorator(_for_app)
 
 
+def _xdg_dir(env, default):
+    """
+    Returns the XDG user directory by first checking the given environment variable
+    and then resulting to the default filepath.
+
+    # Note
+
+    The directory gets created if it doesn't exist.
+    """
+
+    default_xdg_dir = os.path.expanduser(default)
+    dir = os.getenv(env, default_xdg_dir)
+
+    # Ensure the cache_path exists, Python 2 does not have the exist_ok
+    # parameter
+    try:
+        os.makedirs(dir)
+    except OSError:
+        if not os.path.isdir(dir):
+            raise
+
+    return dir
+
```

Comment:
Linting is probably going to expect 2 newlines between module functions.

Reply:
I couldn't check locally as I had trouble setting it up in nixos. would be nice if someone of you could fix this properly.

Reply:
Bull's eyes, @djh82.

@mainrs, you can check the GitHub Action output here: https://github.com/nvbn/thefuck/runs/4701882809?check_suite_focus=true (or clicking below in the "Some checks were not successful" section)

Reply:
Done!